nntrainer/tensor/tensor.cpp

```diff
@@ -633,9 +633,33 @@ Tensor &Tensor::dot(Tensor const &m, Tensor &result, bool trans,
   float *rdata = result.getData();
   const float alpha = 1.0f;
   const float beta = 0.0f;
-
   enum CBLAS_TRANSPOSE transA = trans ? CblasTrans : CblasNoTrans;
   enum CBLAS_TRANSPOSE transB = trans_m ? CblasTrans : CblasNoTrans;
+
+  /// shortcut calculation in case of vector
+  /// case1: (1 * K) X (K * 1)
+  if (M == 1 && N == 1) {
```

Comment:
in this scenario, trans are directly ignored? If yes, then please add that to the documentation of the function.

Reply:
for a vector A (1 * K), A^T == A in terms of memory layout. The case is considered in the test :)

Reply:
Right, `1*k` and `k*1` are same from the perspective from the memory, but in tensor_dim, they are not. Hence, it must be cleared.

Reply:
Ahh, sorry for the misunderstanding, trans are not ignored as trans are resolved to get appropriate (N, M, K) before entering this line. 
Please note that N, M, K denotes a placeholder for (N * K) x (K * M). So case of (1 * K) X (1 * K) is not possible as it should be (1 * K) X (1 * M).

I added some more notes here to clarify.

Reply:
thanks